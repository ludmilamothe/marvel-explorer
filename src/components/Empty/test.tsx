import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import Empty from '.';

const datas = {
  imagePath: '/images/no-search.svg',
  imageAlt:
    'A magnifying glass with sad face and with some bubbles in the background',
  children: 'Some text',
};

describe('<Empty />', () => {
  it('should render the empty correct', () => {
    renderWithProviders(<Empty {...datas} />);

    const image = screen.getByRole('img', { name: datas.imageAlt });
    const children = screen.getByText(datas.children);

    expect(children).toBeInTheDocument();

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', datas.imagePath);
  });

  it('should render the empty styles correct', () => {
    const { container } = renderWithProviders(<Empty {...datas} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
