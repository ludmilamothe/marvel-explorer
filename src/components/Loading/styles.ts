import styled, { css } from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Dot = styled.span`
  ${({ theme }) => css`
    position: relative;
    top: 1.5rem;
    height: 2.4rem;
    width: 2.4rem;
    border-radius: 10rem;
    background-color: ${theme.colors.secondary};
    animation: wave 0.6s ease-in alternate infinite;

    & + & {
      margin-left: ${theme.spacings.xsmall};
    }

    &:nth-child(2) {
      animation-delay: 0.2s;
    }

    &:nth-child(3) {
      animation-delay: 0.4s;
    }

    &:nth-child(4) {
      animation-delay: 0.6s;
    }

    @keyframes wave {
      0% {
        top: 1.5rem;
      }

      100% {
        top: -1.5rem;
      }
    }
  `}
`;
