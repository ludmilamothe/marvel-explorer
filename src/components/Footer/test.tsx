import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/test/helpers';

import Footer from '.';

describe('<Footer />', () => {
  it('should render the footer content correct', () => {
    renderWithTheme(<Footer />);

    const linkText = 'Mauro de Souza';
    const copy = 'Data provided by Marvel. Â© 2020 MARVEL';

    const linkElement = screen.getByRole('link', { name: linkText });

    expect(linkElement).toHaveAttribute(
      'href',
      'https://www.linkedin.com/in/maurodesouzaa/'
    );

    expect(linkElement).toHaveAttribute('target', '_blank');

    expect(screen.getByText(copy)).toBeInTheDocument();
  });

  it('should render the footer styles correct', () => {
    const { container } = renderWithTheme(<Footer />);

    const footer = container.firstChild;

    expect(footer).toMatchSnapshot();
  });
});
