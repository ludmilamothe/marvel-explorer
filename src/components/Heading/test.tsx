import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/test/helpers';

import theme from 'styles/theme/default';
import Heading from '.';

describe('<Heading />', () => {
  it('should render the heading with children', () => {
    const text = 'Just a text to test';

    renderWithTheme(<Heading>{text}</Heading>);

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toBeInTheDocument();

    expect(heading).toMatchInlineSnapshot(`
      .c0 {
        color: #FFF;
        font-weight: 900;
        font-size: 2.4rem;
        max-width: 55rem;
        text-align: center;
      }

      @media (min-width:768px) {
        .c0 {
          font-size: 4.0rem;
        }
      }

      <h1
        class="c0"
      >
        Just a text to test
      </h1>
    `);
  });

  it('should render the heading with correct styles', () => {
    const text = 'Just a text to test';

    renderWithTheme(<Heading>{text}</Heading>);

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyle({
      fontSize: theme.font.sizes.large,
      fontWeight: theme.font.xbold,
      color: theme.colors.white,
    });

    expect(heading).toHaveStyleRule('font-size', theme.font.sizes.xlarge, {
      media: '(min-width: 768px)',
    });
  });
});
