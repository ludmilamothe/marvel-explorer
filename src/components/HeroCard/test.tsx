/* eslint-disable @typescript-eslint/no-var-requires */

import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { storageKey } from 'hooks/useFavoriteHeroes';
import HeroCard from '.';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter');
const push = jest.fn();

useRouter.mockImplementation(() => ({
  push,
}));

const heroData = {
  heroId: 123,
  heroName: 'Iron man',
  heroThumbnail:
    'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55.jpg',
};

describe('<HeroCard />', () => {
  beforeEach(() => {
    window.localStorage.clear();
  });

  it('should render the hero card correct', () => {
    renderWithProviders(<HeroCard {...heroData} />);

    expect(
      screen.getByRole('img', { name: heroData.heroName })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: heroData.heroName })
    ).toBeInTheDocument();

    expect(screen.getByLabelText(/add hero to favorites/i)).toBeInTheDocument();
  });

  it('should render the hero card styles correct', () => {
    const { container } = renderWithProviders(<HeroCard {...heroData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add hero to favorites', () => {
    renderWithProviders(<HeroCard {...heroData} />);

    const addToFavoriteElement = screen.getByLabelText(
      /add hero to favorites/i
    );

    expect(addToFavoriteElement).toBeInTheDocument();
    fireEvent.click(addToFavoriteElement);

    const removeFromFavoriteElement = screen.getByLabelText(
      /remove hero to favorites/i
    );

    expect(removeFromFavoriteElement).toBeInTheDocument();
    expect(addToFavoriteElement).not.toBeInTheDocument();

    const storagedHero = JSON.parse(window.localStorage.getItem(storageKey)!);

    expect(storagedHero).toEqual([heroData]);
  });

  it('should remove hero to favorites', () => {
    window.localStorage.setItem(storageKey, JSON.stringify([heroData]));

    renderWithProviders(<HeroCard {...heroData} />);

    const removeFromFavoriteElement = screen.getByLabelText(
      /remove hero to favorites/i
    );

    expect(removeFromFavoriteElement).toBeInTheDocument();
    fireEvent.click(removeFromFavoriteElement);

    const addToFavoriteElement = screen.getByLabelText(
      /add hero to favorites/i
    );

    expect(removeFromFavoriteElement).not.toBeInTheDocument();
    expect(addToFavoriteElement).toBeInTheDocument();

    const storagedHero = JSON.parse(window.localStorage.getItem(storageKey)!);
    expect(storagedHero).toEqual([]);
  });

  it('should navigate to hero page', () => {
    const { container } = renderWithProviders(<HeroCard {...heroData} />);

    const heroCard = container.firstChild!;
    fireEvent.click(heroCard);

    expect(push).toBeCalledTimes(1);
    expect(push).toBeCalledWith(`/characters/${heroData.heroId}`);
  });
});
