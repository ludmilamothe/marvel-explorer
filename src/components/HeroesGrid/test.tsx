import { renderWithProviders } from 'utils/test/helpers';

import HeroesGrid from '.';

describe('<HeroesGrid />', () => {
  it('should render the HeroesGrid styles correct', () => {
    const { container } = renderWithProviders(<HeroesGrid />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
