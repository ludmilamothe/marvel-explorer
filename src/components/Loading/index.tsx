import * as S from './styles';

const Loading = () => (
  <S.Container>
    <S.Dot />
    <S.Dot />
    <S.Dot />
    <S.Dot />
  </S.Container>
);

export default Loading;
