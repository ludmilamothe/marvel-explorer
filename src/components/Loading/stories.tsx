import { Meta, Story } from '@storybook/react/types-6-0';

import Loading from '.';

export default {
  title: 'Loading',
  component: Loading,
} as Meta;

export const Basic: Story = args => <Loading {...args} />;
