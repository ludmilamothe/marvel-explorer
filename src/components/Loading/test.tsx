import { renderWithProviders } from 'utils/test/helpers';

import Loading from '.';

describe('<Loading />', () => {
  it('should render the Loading', () => {
    const { container } = renderWithProviders(<Loading />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
