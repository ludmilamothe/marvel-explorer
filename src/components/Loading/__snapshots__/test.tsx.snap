// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the Loading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: relative;
  top: 1.5rem;
  height: 2.4rem;
  width: 2.4rem;
  border-radius: 10rem;
  background-color: #C6A972;
  -webkit-animation: wave 0.6s ease-in alternate infinite;
  animation: wave 0.6s ease-in alternate infinite;
}

.c1 + .c1 {
  margin-left: 1.2rem;
}

.c2:nth-child(2) {
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.c2:nth-child(3) {
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
}

.c2:nth-child(4) {
  -webkit-animation-delay: 0.6s;
  animation-delay: 0.6s;
}

<div
  class="c0"
>
  <span
    class="c1 c2"
  />
  <span
    class="c1 c2"
  />
  <span
    class="c1 c2"
  />
  <span
    class="c1 c2"
  />
</div>
`;
